subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -156,34 +175,89 @@ void bl_boot(const struct fw_info *fw_info)
 	__set_MSP(vector_table[0]);
 	__set_PSP(0);
 
-#if CONFIG_SB_CLEANUP_RAM
 	__asm__ volatile (
 		/* vector_table[1] -> r0 */
-		"   mov r0, %0\n"
+		"   mov  r0, %0\n"
+#ifdef CONFIG_SB_CLEANUP_RAM
 		/* Base to write -> r1 */
-		"   mov r1, %1\n"
+		"   mov  r1, %1\n"
 		/* Size to write -> r2 */
-		"   mov r2, %2\n"
+		"   mov  r2, %2\n"
 		/* Value to write -> r3 */
-		"   mov r3, %3\n"
+		"   mov  r3, %3\n"
 		"clear:\n"
-		"   str r3, [r1]\n"
-		"   add r1, r1, #4\n"
-		"   sub r2, r2, #4\n"
-		"   cbz r2, out\n"
-		"   b   clear\n"
+		"   str  r3, [r1]\n"
+		"   add  r1, r1, #4\n"
+		"   sub  r2, r2, #4\n"
+		"   cbz  r2, out\n"
+		"   b    clear\n"
 		"out:\n"
 		"   dsb\n"
+#endif /* CONFIG_SB_CLEANUP_RAM */
+
+#ifdef CONFIG_SB_DISABLE_SELF_RWX
+		/* FUNCTION_BUFFER_LEN */
+		"   mov  r4, %4\n"
+		/* ram_exec_buf */
+		"   mov  r2, %5\n"
+		"   movw r3, :lower16:bootconf_disable_rwx\n"
+		"   movt r3, :upper16:bootconf_disable_rwx\n"
+		/* adjust address for thumb */
+		"   and  r3, #0xfffffffe\n"
```

Comment:
thumb instruction should be +1 afaik

Reply:
That's below, here is adjustment to ensure that start of the copy isn't cropped